test/expr/unary/keypath/keypath.swift

```diff
@@ -229,10 +229,12 @@ func testDisembodiedStringInterpolation(x: Int) {
 }
 
 func testNoComponents() {
-  let _: KeyPath<A, A> = \A // expected-error{{must have at least one component}}
+  let _: KeyPath<A, A> = \A // expected-error{{must have at least one component}} 
+  // expected-note@+2 {{arguments to generic parameter 'Root' ('Root' and 'C<T>') are expected to be equal}}
+  // expected-note@+1 {{arguments to generic parameter 'Value' ('Value' and 'A') are expected to be equal}}
   let _: KeyPath<C, A> = \C // expected-error{{must have at least one component}} expected-error{{}}
   // expected-error@-1 {{generic parameter 'T' could not be inferred}}
-  // expected-error@-2 {{cannot convert value of type 'KeyPath<Root, Value>' to specified type 'KeyPath<C<T>, A>'}}
+  // expected-error@-2 {{cannot assign value of type 'KeyPath<Root, Value>' to type 'KeyPath<C<T>, A>'}}
```

Comment:
`it seems like the only failure here should be `generic parameter 'T' could not be inferred` right?

Reply:
Yes, I'll note that to try to take a look at this once we finish this one :)